packages/protocol/contracts/tokenvault/BridgedERC20.sol

```diff
@@ -70,22 +79,66 @@ contract BridgedERC20 is
         srcDecimals = _decimals;
     }
 
+    function startOutboundMigration(address to)
+        external
+        nonReentrant
+        whenNotPaused
+        onlyFromNamed("erc20_vault")
+    {
+        if (migratingTo != address(0)) revert BRIDGED_TOKEN_PERMISSION_DENIED();
+        if (to == address(0)) revert BRIDGED_TOKEN_INVALID_PARAMS();
+
+        migratingTo = to;
+        emit Migration(migratingFrom, migratingTo);
+    }
+
+    function startInboundMigration(address from)
+        external
+        nonReentrant
+        whenNotPaused
+        onlyFromNamed("erc20_vault")
+    {
+        if (migratingFrom != address(0)) revert BRIDGED_TOKEN_PERMISSION_DENIED();
+        if (from == address(0)) revert BRIDGED_TOKEN_INVALID_PARAMS();
+
+        migratingFrom = from;
+        emit Migration(migratingFrom, migratingTo);
+    }
+
+    function stopInboundMigration() external nonReentrant whenNotPaused onlyOwner {
+        if (migratingFrom == address(0)) revert BRIDGED_TOKEN_PERMISSION_DENIED();
+        migratingFrom = address(0);
+        emit Migration(migratingFrom, migratingTo);
+    }
+
     /// @notice Mints tokens to an account.
-    /// @dev Only an ERC20Vault can call this function.
     /// @param account The account to mint tokens to.
     /// @param amount The amount of tokens to mint.
-    function mint(address account, uint256 amount) public onlyFromNamed("erc20_vault") {
+    function mint(address account, uint256 amount) public nonReentrant whenNotPaused {
+        if (migratingFrom == address(0)) {
+            if (msg.sender != resolve("erc20_vault", true)) revert RESOLVER_DENIED();
+            emit Transfer(address(0), account, amount);
```

Comment:
> Sorry I don't get it, can you elaborate a bit more?

Just meant, the `emit Transfer() `event is a duplicate if we emit it here too, because the `_mint()` function emits it already.

Reply:
Fixed